lib/src/provider.dart

```diff
@@ -421,7 +421,8 @@ class ProviderNotFoundException implements Exception {
     return '''
 Error: Could not find the correct Provider<$valueType> above this $widgetType Widget
 
-This likely happens because you used a `BuildContext` that does not include the provider
+This likely happens because you created a new provider but did not performed a 'Hot Restart'
```

Comment:
I'd instead include that in the "common scenarios" list, making sure to explain that this is for when adding providers inside the `main` function